Src/Support/Google.Apis.Auth/OAuth2/ComputeCredential.cs

```diff
@@ -461,5 +492,55 @@ async Task<bool> IsLinuxGoogleBiosAsync()
                 return productName == "Google" || productName == "Google Compute Engine";
             }
         }
+
+        private async static Task<string> GetComputeEngineUniverseDomainUncachedAsync()
+        {
+            Logger.Info("Attempting to fetch the universe domain from the metadata server.");
+
+            // Using the built-in HttpClient, as we want bare bones functionality - we'll control retries.
+            // Use the same one across all attempts, which may contribute to speedier retries.
+            using (var httpClient = new HttpClient())
+            {
+                int attempts = 0;
+                // We use the same timeouts as we do for token fetching.
+                foreach (var timeout in TokenRefreshManager.RefreshTimeouts)
+                {
+                    attempts++;
+                    var cts = new CancellationTokenSource(timeout);
+                    HttpResponseMessage response = null;
+                    try
+                    {
+                        var httpRequest = new HttpRequestMessage(HttpMethod.Get, GoogleAuthConsts.EffectiveComputeUniverDomainUrl);
+                        httpRequest.Headers.Add(MetadataFlavor, GoogleMetadataHeader);
+
+                        response = await httpClient.SendAsync(httpRequest, cts.Token).ConfigureAwait(false);
+                        response.EnsureSuccessStatusCode();
+
+                        return await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                    }
+                    catch (Exception) when (response?.StatusCode == HttpStatusCode.NotFound)
```

Comment:
Passing by comment ) 

what if status code is not 404? technically possible.. it will fall into the InvalidOperaition?

Reply:
In that case, and if it's not a timeout (timeout is handled by catching the OperationCanceldException) then the original exception is bubled up. This is also what happens on token fetching, and I'm mimicking that here as per spec.
Does that sound good?

Reply:
Oh got it.. I misunderstood the Exception catch the first time. Looks good.